--- configure_org	2024-06-16 17:27:54.000000000 +0200
+++ configure		2024-09-26 12:04:24.475108676 +0200
@@ -3493,6 +3493,7 @@
 RPATH_CANDIDATE_REAL_DIRS=''
 RPATH_CANDIDATE_DIRS=''
 
+if false; then
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: Searching for libraries" >&5
 printf "%s\n" "Searching for libraries" >&6; }
 
@@ -3532,7 +3533,7 @@
 fi
 
 done
-
+fi
 
 # Checks for programs.
 
