From e97deaa1bf7a9f452e7762787a7259a3a167d14a Mon Sep 17 00:00:00 2001
From: Jean-Baptiste Kempf <jb@videolan.org>
Date: Sat, 28 Feb 2015 19:38:04 +0000
Subject: [PATCH 08/11] Fix C++ compilation using the library

The linker wouldn't find the symbol otherwise

 100.0% src/dvbcsa/
diff --git a/src/dvbcsa/dvbcsa.h b/src/dvbcsa/dvbcsa.h
index e051f4f..b4d29d1 100644
--- a/src/dvbcsa/dvbcsa.h
+++ b/src/dvbcsa/dvbcsa.h
@@ -27,6 +27,10 @@
 #ifndef LIBDVBCSA_H_
 #define LIBDVBCSA_H_
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /* csa control word */
 typedef unsigned char		dvbcsa_cw_t[8];
 
@@ -108,5 +112,9 @@ void dvbcsa_bs_encrypt(const struct dvbcsa_bs_key_s *key,
 		       const struct dvbcsa_bs_batch_s *pcks,
 		       unsigned int maxlen);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif
 
-- 
2.43.0

