--- avm/config-airoha01-5699_08.25
+++ freetz/config-airoha01-5699_08.25
@@ -4,12 +4,11 @@
 #
 
 #
-# Compiler: aarch64-buildroot-linux-musl-gcc.br_real (Buildroot -g91c9c7f8) 10.5.0
+# Compiler: aarch64-unknown-linux-gnu-gcc (GCC) 10.5.0
 #
 CONFIG_CC_IS_GCC=y
 CONFIG_GCC_VERSION=100500
 CONFIG_CLANG_VERSION=0
-CONFIG_CC_CAN_LINK=y
 CONFIG_CC_HAS_ASM_GOTO=y
 CONFIG_CC_HAS_ASM_INLINE=y
 CONFIG_IRQ_WORK=y
