--- avm/config-maple-1249_08.20
+++ freetz/config-maple-1249_08.20
@@ -4,12 +4,11 @@
 #
 
 #
-# Compiler: arm-buildroot-linux-musleabi-gcc.br_real (Buildroot -g162062dc) 8.4.0
+# Compiler: arm-unknown-linux-gnueabi-gcc (GCC) 8.4.0
 #
 CONFIG_CC_IS_GCC=y
 CONFIG_GCC_VERSION=80400
 CONFIG_CLANG_VERSION=0
-CONFIG_CC_CAN_LINK=y
 CONFIG_CC_HAS_ASM_GOTO=y
 CONFIG_CC_HAS_ASM_INLINE=y
 CONFIG_IRQ_WORK=y
@@ -665,8 +664,9 @@
 CONFIG_ARCH_HAS_GCOV_PROFILE_ALL=y
 # end of GCOV-based kernel profiling
 
-CONFIG_PLUGIN_HOSTCC=""
+CONFIG_PLUGIN_HOSTCC="g++"
 CONFIG_HAVE_GCC_PLUGINS=y
+# CONFIG_GCC_PLUGINS is not set
 # end of General architecture-dependent options
 
 CONFIG_RT_MUTEXES=y
@@ -752,7 +752,6 @@
 CONFIG_GPIOLIB_AVM_EXTENSION=y
 # CONFIG_AVM_XHCI_BULK2INT is not set
 CONFIG_AVM_RTE=y
-# CONFIG_AVM_GENERIC_RTE is not set
 CONFIG_AVM_WAR_FIT_PHYS_OFS=0x40000000
 CONFIG_SHOW_NOTIFIER_IF_OOPS=y
 CONFIG_QCA_NSS_AVM_FRITZ_BOX=y
@@ -3007,6 +3006,7 @@
 # CONFIG_SENSORS_ASC7621 is not set
 # CONFIG_SENSORS_ASPEED is not set
 # CONFIG_SENSORS_ATXP1 is not set
+# CONFIG_SENSORS_AVM_GPIO_POWERMON is not set
 # CONFIG_SENSORS_DRIVETEMP is not set
 # CONFIG_SENSORS_DS620 is not set
 # CONFIG_SENSORS_DS1621 is not set
