--- avm/config-miami-7690_07.61
+++ freetz/config-miami-7690_07.61
@@ -4,12 +4,11 @@
 #
 
 #
-# Compiler: arm-buildroot-linux-musleabi-gcc.br_real (Buildroot 2020.02.1-gf277c724) 8.4.0
+# Compiler: arm-unknown-linux-gnueabi-gcc (GCC) 8.4.0
 #
 CONFIG_CC_IS_GCC=y
 CONFIG_GCC_VERSION=80400
 CONFIG_CLANG_VERSION=0
-CONFIG_CC_CAN_LINK=y
 CONFIG_CC_HAS_ASM_GOTO=y
 CONFIG_CC_HAS_ASM_INLINE=y
 CONFIG_IRQ_WORK=y
@@ -693,8 +692,9 @@
 CONFIG_ARCH_HAS_GCOV_PROFILE_ALL=y
 # end of GCOV-based kernel profiling
 
-CONFIG_PLUGIN_HOSTCC=""
+CONFIG_PLUGIN_HOSTCC="g++"
 CONFIG_HAVE_GCC_PLUGINS=y
+# CONFIG_GCC_PLUGINS is not set
 # end of General architecture-dependent options
 
 CONFIG_RT_MUTEXES=y
