--- src/auto/configure
+++ src/auto/configure
@@ -14156,6 +14156,7 @@
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.beam conftest.$ac_ext
 
+if false; then
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for timer_create without -lrt" >&5
 printf %s "checking for timer_create without -lrt... " >&6; }
 if test ${vim_cv_timer_create+y}
@@ -14289,6 +14290,7 @@
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
 fi
+fi
 
 fi
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $vim_cv_timer_create_works" >&5
