--- lib/CMakeLists.txt
+++ lib/CMakeLists.txt
@@ -60,7 +60,10 @@
 	util_mosq.c util_topic.c util_mosq.h
 	will_mosq.c will_mosq.h)
 
-set (LIBRARIES OpenSSL::SSL)
+set (LIBRARIES "")
+if (WITH_TLS)
+set (LIBRARIES ${LIBRARIES} OpenSSL::SSL)
+endif (WITH_TLS)
 
 if (UNIX AND NOT APPLE AND NOT ANDROID)
 	find_library(LIBRT rt)
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -163,7 +163,9 @@
     set (MOSQ_LIBS ${MOSQ_LIBS} ${DLT_LIBRARIES})
 endif (WITH_DLT)
 
+if (WITH_TLS)
 set (MOSQ_LIBS ${MOSQ_LIBS} OpenSSL::SSL)
+endif (WITH_TLS)
 # Check for getaddrinfo_a
 include(CheckLibraryExists)
 check_library_exists(anl getaddrinfo_a  "" HAVE_GETADDRINFO_A)
