--- usr/www/all/internet/dsl_overview.lua
+++ usr/www/all/internet/dsl_overview.lua
@@ -200,9 +200,7 @@
 local margeSend=tonumber(box.query("sar:settings/UsNoiseBits"))or-1
 local rfi=tonumber(box.query("sar:settings/RFI_mode"))or-1
 local inp=tonumber(box.query("sar:settings/DsINP"))or-1
-if margeReceive>0 or margeSend>0 or rfi>0 or inp>0 then
-return true
-end
+return false
 end
 local function writeUmtsBlock()
 if config.USB_GSM then
