--- configure
+++ configure
@@ -4774,6 +4774,7 @@
 
 config_appended_defs="$config_appended_defs${newline}INLINE=${INLINE}"
 
+if false; then
 # which switch to pass runtime path to dynamic libraries to the linker
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if linker supports -R" >&5
 printf %s "checking if linker supports -R... " >&6; }
@@ -4928,6 +4929,7 @@
 fi
 
 config_appended_defs="$config_appended_defs${newline}CC_LD_DYNPATH=${CC_LD_DYNPATH}"
+fi
 #AC_PROG_INSTALL		# needs install-sh or install.sh in sources
 if test -n "$ac_tool_prefix"; then
   for ac_prog in gar ar
--- shared.mak
+++ shared.mak
@@ -114,7 +114,7 @@
 ## With CC_LD_DYNPATH set to either an empty string or to "-L", the
 ## the directory is not shown the second time.
 define libpath_template
--L$(1) $(if $(filter-out -L,$(CC_LD_DYNPATH)),$(CC_LD_DYNPATH)$(1))
+-L$(1)
 endef
 
 # Populate build information into a file via GIT-VERSION-GEN. Requires the
