--- Makefile
+++ Makefile
@@ -43,9 +43,9 @@
 HOSTOS := $(shell uname -s | tr '[:upper:]' '[:lower:]' | \
 	    sed -e 's/\(cygwin\|msys\).*/\1/')
 
-NO_PYTHON ?= 0
+NO_PYTHON := 1
 
-NO_VALGRIND := $(shell $(PKG_CONFIG) --exists valgrind; echo $$?)
+NO_VALGRIND := 1
 ifeq ($(NO_VALGRIND),1)
 	CPPFLAGS += -DNO_VALGRIND
 else
@@ -56,7 +56,7 @@
 # both variants would require either cpp magic or passing
 # -Wno-error=discarded-qualifiers to the compiler. For the sake of simplicity
 # just support libyaml >= 0.2.3.
-NO_YAML := $(shell $(PKG_CONFIG) --atleast-version 0.2.3 yaml-0.1; echo $$?)
+NO_YAML := 1
 ifeq ($(NO_YAML),1)
 	CFLAGS += -DNO_YAML
 else
