--- scripts/kconfig/symbol.c
+++ scripts/kconfig/symbol.c
@@ -374,7 +374,7 @@
 	struct gstr gs = str_new();
 
 	str_printf(&gs,
-		   "\nWARNING: unmet direct dependencies detected for %s\n",
+		   "\nERROR: unmet direct dependencies detected for %s\n",
 		   sym->name);
 	str_printf(&gs,
 		   "  Depends on [%c]: ",
@@ -390,6 +390,7 @@
 	fputs(str_get(&gs), stderr);
 	str_free(&gs);
 	sym_warnings++;
+	exit(1);
 }
 
 bool sym_dep_errors(void)
