--- scripts/kconfig/mconf.c
+++ scripts/kconfig/mconf.c
@@ -627,9 +627,9 @@
 		dialog_clear();
 		res = dialog_checklist(prompt ? prompt : "Main Menu",
 					radiolist_instructions,
-					MENUBOX_HEIGHT_MIN,
+					getmaxy(stdscr) - CHECKLIST_HEIGHT_MIN,
 					MENUBOX_WIDTH_MIN,
-					CHECKLIST_HEIGHT_MIN);
+					getmaxy(stdscr) - MENUBOX_HEIGHT_MIN);
 		selected = item_activate_selected();
 		switch (res) {
 		case 0:
