--- configure
+++ configure
@@ -30584,15 +30584,10 @@
 printf "%s\n" "and out again" >&6; }
 
 echo $ECHO_N "ordering CD from https://tobi.oetiker.ch/wish $ECHO_C" 1>&6
-sleep 1
 echo $ECHO_N ".$ECHO_C" 1>&6
-sleep 1
 echo $ECHO_N ".$ECHO_C" 1>&6
-sleep 1
 echo $ECHO_N ".$ECHO_C" 1>&6
-sleep 1
 echo $ECHO_N ".$ECHO_C" 1>&6
-sleep 1
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result:  just kidding ;-)" >&5
 printf "%s\n" " just kidding ;-)" >&6; }
 echo
