--- configure
+++ configure
@@ -7325,7 +7325,7 @@
 
 
 
-for flag in -fPIE; do
++false && for flag in -fPIE; do
   as_CACHEVAR=`printf "%s\n" "ax_cv_check_cflags__$flag" | sed "$as_sed_sh"`
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking whether C compiler accepts $flag" >&5
 printf %s "checking whether C compiler accepts $flag... " >&6; }
@@ -7663,7 +7663,7 @@
 
 
 
-for flag in -fPIE -pie; do
++false && for flag in -fPIE -pie; do
   as_CACHEVAR=`printf "%s\n" "ax_cv_check_ldflags__$flag" | sed "$as_sed_sh"`
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking whether the linker accepts $flag" >&5
 printf %s "checking whether the linker accepts $flag... " >&6; }
