--- src/fs.c
+++ src/fs.c
@@ -179,7 +179,11 @@
 		return;
 	}
 
+#ifdef O_CLOEXEC
 	if ((dir_fd = open(dir, O_RDONLY | O_CLOEXEC)) == -1)
+#else
+	if ((dir_fd = open(dir, O_RDONLY)) == -1)
+#endif
 		return;
 	if (fstat(dir_fd, &statbuf)!=0) {
 		close(dir_fd);
@@ -210,7 +214,11 @@
 			continue;
 		}
 		snprintf(entryname, 512, "%s/%s", dir, di->d_name);
+#ifdef O_CLOEXEC
 		if ((file_fd = open(entryname, O_RDONLY | O_CLOEXEC)) == -1)
+#else
+		if ((file_fd = open(entryname, O_RDONLY)) == -1)
+#endif
 			continue;
 		if (fstat(file_fd, &statbuf)!=0) {
 			close(file_fd);
