--- libssh2.pc.in
+++ libssh2.pc.in
@@ -14,8 +14,6 @@
 URL: https://libssh2.org/
 Description: Library for SSH-based communication
 Version: @LIBSSH2_VERSION@
 Requires: @LIBSSH2_PC_REQUIRES@
-Requires.private: @LIBSSH2_PC_REQUIRES_PRIVATE@
-Libs: -L${libdir} -lssh2 @LIBSSH2_PC_LIBS@
-Libs.private: @LIBSSH2_PC_LIBS_PRIVATE@
+Libs: -L${libdir} -lssh2 @LIBGCRYPT@ @LIBMBEDCRYPTO@
 Cflags: -I${includedir}
