--- doc/hlp/Makefile.am
+++ doc/hlp/Makefile.am
@@ -1,4 +1,7 @@
-DIST_SUBDIRS = es hu it pl ru sr
+DIST_SUBDIRS =
+if USE_NLS
+DIST_SUBDIRS += es hu it pl ru sr
+endif
 if USE_NLS
 SUBDIRS = $(DOC_LINGUAS)
 endif
--- doc/hlp/Makefile.in
+++ doc/hlp/Makefile.in
@@ -471,7 +471,8 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-DIST_SUBDIRS = es hu it pl ru sr
+DIST_SUBDIRS =
+@USE_NLS_TRUE@DIST_SUBDIRS += es hu it pl ru sr
 @USE_NLS_TRUE@SUBDIRS = $(DOC_LINGUAS)
 hlpdir = $(pkgdatadir)/help
 hlp_DATA = mc.hlp
--- doc/man/Makefile.am
+++ doc/man/Makefile.am
@@ -1,4 +1,7 @@
-DIST_SUBDIRS = es hu it pl ru sr
+DIST_SUBDIRS =
+if USE_NLS
+DIST_SUBDIRS += es hu it pl ru sr
+endif
 if USE_NLS
 SUBDIRS = $(DOC_LINGUAS)
 endif
--- doc/man/Makefile.in
+++ doc/man/Makefile.in
@@ -475,7 +475,8 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-DIST_SUBDIRS = es hu it pl ru sr
+DIST_SUBDIRS =
+@USE_NLS_TRUE@DIST_SUBDIRS += es hu it pl ru sr
 @USE_NLS_TRUE@SUBDIRS = $(DOC_LINGUAS)
 man_MANS = mc.1 mcview.1 $(am__append_1) $(am__append_2)
 CLEANFILES = $(man_MANS)
@@ -748,7 +749,8 @@
 	for dir in "$(DESTDIR)$(man1dir)"; do \
 	  test -z "$$dir" || $(MKDIR_P) "$$dir"; \
 	done
-install: install-recursive
+install:
+#install: install-recursive
 install-exec: install-exec-recursive
 install-data: install-data-recursive
 uninstall: uninstall-recursive
--- src/vfs/extfs/helpers/Makefile.am
+++ src/vfs/extfs/helpers/Makefile.am
@@ -1,7 +1,7 @@
 extfsdir = $(EXTFSHELPERSDIR)
 
 # Files to install and distribute other than extfs scripts
-EXTFS_MISC  = README README.extfs
+EXTFS_MISC  =
 
 # Scripts hat don't need adaptation to the local system
 EXTFS_CONST = bpp changesetfs gitfs+ patchsetfs rpm trpm u7z
--- src/vfs/extfs/helpers/Makefile.in
+++ src/vfs/extfs/helpers/Makefile.in
@@ -431,7 +431,7 @@
 extfsdir = $(EXTFSHELPERSDIR)
 
 # Files to install and distribute other than extfs scripts
-EXTFS_MISC = README README.extfs
+EXTFS_MISC =
 
 # Scripts hat don't need adaptation to the local system
 EXTFS_CONST = bpp changesetfs gitfs+ patchsetfs rpm trpm u7z
--- src/vfs/shell/helpers/Makefile.am
+++ src/vfs/shell/helpers/Makefile.am
@@ -1,7 +1,7 @@
 shelldir = $(libexecdir)/@PACKAGE@/shell
 
 # Files to install and distribute other than shell scripts
-SHELL_MISC  = README.shell
+SHELL_MISC  =
 
 # Install and distribute SHELL helper scripts w/o shebang & executable bit as data
 shell_DATA = $(SHELL_MISC) ls mkdir fexists unlink chown chmod rmdir ln mv hardlink get send append info utime
--- src/vfs/shell/helpers/Makefile.in
+++ src/vfs/shell/helpers/Makefile.in
@@ -417,7 +417,7 @@
 shelldir = $(libexecdir)/@PACKAGE@/shell
 
 # Files to install and distribute other than shell scripts
-SHELL_MISC = README.shell
+SHELL_MISC =
 
 # Install and distribute SHELL helper scripts w/o shebang & executable bit as data
 shell_DATA = $(SHELL_MISC) ls mkdir fexists unlink chown chmod rmdir ln mv hardlink get send append info utime
