--- Makefile.am
+++ Makefile.am
@@ -86,8 +86,8 @@
 
 bin_SCRIPTS = curl-config
 
-SUBDIRS = lib docs src scripts
-DIST_SUBDIRS = $(SUBDIRS) tests packages scripts include docs
+SUBDIRS = lib      src scripts
+DIST_SUBDIRS = $(SUBDIRS)       packages scripts include
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -201,14 +201,11 @@
 else
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
 endif
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	(cd include && $(MAKE) uninstall)
-	(cd docs && $(MAKE) uninstall)
-	(cd docs/libcurl && $(MAKE) uninstall)
 
 ca-bundle: $(srcdir)/scripts/mk-ca-bundle.pl
 	@echo "generating a fresh ca-bundle.crt"
--- Makefile.in
+++ Makefile.in
@@ -621,8 +621,8 @@
  $(VC14_30_LIBVCXPROJ) $(VC14_30_SRCVCXPROJ)
 
 bin_SCRIPTS = curl-config
-SUBDIRS = lib docs src scripts
-DIST_SUBDIRS = $(SUBDIRS) tests packages scripts include docs
+SUBDIRS = lib      src scripts
+DIST_SUBDIRS = $(SUBDIRS)       packages scripts include
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
 LIB_VAUTH_CFILES = \
@@ -1755,13 +1755,10 @@
 @BUILD_DOCS_TRUE@	(cd docs/libcurl && $(MAKE) install)
 @BUILD_DOCS_FALSE@install-data-hook:
 @BUILD_DOCS_FALSE@	(cd include && $(MAKE) install)
-@BUILD_DOCS_FALSE@	(cd docs && $(MAKE) install)
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	(cd include && $(MAKE) uninstall)
-	(cd docs && $(MAKE) uninstall)
-	(cd docs/libcurl && $(MAKE) uninstall)
 
 ca-bundle: $(srcdir)/scripts/mk-ca-bundle.pl
 	@echo "generating a fresh ca-bundle.crt"
