--- curl-config.in
+++ curl-config.in
@@ -27,6 +27,7 @@
 # Used in @libdir@
 # shellcheck disable=SC2034
 exec_prefix=@exec_prefix@
+libdir=@libdir@
 # shellcheck disable=SC2034
 includedir=@includedir@
 cppflag_curl_staticlib=@CPPFLAG_CURL_STATICLIB@
@@ -153,8 +154,8 @@
     ;;
 
   --libs)
-    if test "X@libdir@" != "X/usr/lib" -a "X@libdir@" != "X/usr/lib64"; then
-      CURLLIBDIR="-L@libdir@ "
+    if test "X${libdir}" != "X/usr/lib" -a "X${libdir}" != "X/usr/lib64"; then
+      CURLLIBDIR="-L${libdir} "
     else
       CURLLIBDIR=""
     fi
@@ -171,7 +172,7 @@
 
   --static-libs)
     if test "X@ENABLE_STATIC@" != "Xno" ; then
-      echo "@libdir@/libcurl.@libext@" @LDFLAGS@ @LIBCURL_LIBS@
+      echo "${libdir}/libcurl.@libext@" @LDFLAGS@ @LIBCURL_LIBS@
     else
       echo 'curl was built with static libraries disabled' >&2
       exit 1
