--- libtransmission/session.h
+++ libtransmission/session.h
@@ -438,7 +438,7 @@
         bool seed_queue_enabled = false;
         bool sequential_download = false;
         bool should_delete_source_torrents = false;
-        bool should_scrape_paused_torrents = true;
+        bool should_scrape_paused_torrents = false;
         bool should_start_added_torrents = true;
         bool speed_limit_down_enabled = false;
         bool speed_limit_up_enabled = false;
