--- third-party/dht/CMakeLists.txt
+++ third-party/dht/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.5)
 project(dht C)
 
 add_library(${PROJECT_NAME} STATIC
--- third-party/libnatpmp/CMakeLists.txt
+++ third-party/libnatpmp/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.5)
 project(natpmp C)
 
 add_definitions(-DNATPMP_STATICLIB -DENABLE_STRNATPMPERR)
