--- Makefile.am
+++ Makefile.am
@@ -87,8 +87,8 @@
 
 bin_SCRIPTS = curl-config
 
-SUBDIRS = lib docs src scripts
-DIST_SUBDIRS = $(SUBDIRS) tests packages include docs
+SUBDIRS = lib      src scripts
+DIST_SUBDIRS = $(SUBDIRS)       packages include
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -189,14 +189,11 @@
 else
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
 endif
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	(cd include && $(MAKE) uninstall)
-	(cd docs && $(MAKE) uninstall)
-	(cd docs/libcurl && $(MAKE) uninstall)
 
 ca-bundle: $(srcdir)/scripts/mk-ca-bundle.pl
 	@echo "generating a fresh ca-bundle.crt"
--- Makefile.in
+++ Makefile.in
@@ -500,8 +500,8 @@
 
 DISTCLEANFILES = buildinfo.txt
 bin_SCRIPTS = curl-config
-SUBDIRS = lib docs src scripts
-DIST_SUBDIRS = $(SUBDIRS) tests packages include docs
+SUBDIRS = lib      src scripts
+DIST_SUBDIRS = $(SUBDIRS)       packages include
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
 all: all-recursive
@@ -1140,13 +1140,10 @@
 @BUILD_DOCS_TRUE@	(cd docs/libcurl && $(MAKE) install)
 @BUILD_DOCS_FALSE@install-data-hook:
 @BUILD_DOCS_FALSE@	(cd include && $(MAKE) install)
-@BUILD_DOCS_FALSE@	(cd docs && $(MAKE) install)
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	(cd include && $(MAKE) uninstall)
-	(cd docs && $(MAKE) uninstall)
-	(cd docs/libcurl && $(MAKE) uninstall)
 
 ca-bundle: $(srcdir)/scripts/mk-ca-bundle.pl
 	@echo "generating a fresh ca-bundle.crt"
