--- misc/Makefile.in
+++ misc/Makefile.in
@@ -18,6 +18,8 @@
 @LINUX_CMT@E4CRYPT_PROG = e4crypt
 @LINUX_CMT@E4CRYPT_MAN= e4crypt.8
 
+@DEFRAG_CMT@@LINUX_CMT@E4DEFRAG_PROG_PROFILED= e4defrag.profiled
+
 @IMAGER_CMT@E2IMAGE_PROG= e2image
 @IMAGER_CMT@E2IMAGE_STATIC= e2image.static
 @IMAGER_CMT@E2IMAGE_MAN= e2image.8
@@ -147,7 +149,7 @@
 @PROFILE_CMT@all:: tune2fs.profiled blkid.profiled e2image.profiled \
 	e2undo.profiled mke2fs.profiled dumpe2fs.profiled fsck.profiled \
 	logsave.profiled filefrag.profiled uuidgen.profiled $(UUIDD_PROFILED) \
-	e2image.profiled e4defrag.profiled e4crypt.profiled \
+	e2image.profiled $(E4DEFRAG_PROG_PROFILED) e4crypt.profiled \
 	e2freefrag.profiled
 
 profiled:
