--- lib/blkid/Makefile.in
+++ lib/blkid/Makefile.in
@@ -147,9 +147,9 @@
 	$(Q) cd $(top_builddir); CONFIG_FILES=lib/blkid/blkid.pc ./config.status
 
 installdirs::
-	$(E) "	MKDIR_P $(libdir) $(includedir)/blkid"
+	$(E) "	MKDIR_P $(libdir) $(includedir)/blkid $(man3dir)"
 	$(Q) $(MKDIR_P) $(DESTDIR)$(libdir) \
-		$(DESTDIR)$(includedir)/blkid $(DESTDIR)$(pkgconfigdir)
+		$(DESTDIR)$(includedir)/blkid $(DESTDIR)$(pkgconfigdir) $(DESTDIR)$(man3dir)
 
 install:: all installdirs 
 	$(E) "	INSTALL_DATA $(libdir)/libblkid.a"
