--- setup.py
+++ setup.py
@@ -115,7 +115,7 @@
         # Tweak compiler options
         if self.compiler.compiler_type in ('unix', 'cygwin', 'mingw32'):
             # Make assert() statements always work
-            self.__remove_compiler_option("-DNDEBUG")
+#            self.__remove_compiler_option("-DNDEBUG")

             if USE_GCOV:    # TODO - move this to configure.ac
                 self.__add_compiler_option("-fprofile-arcs")
