--- exporters/pkg-config/libcrypto.pc.in
+++ exporters/pkg-config/libcrypto.pc.in
@@ -17,8 +17,7 @@
 Name: OpenSSL-libcrypto
 Description: OpenSSL cryptography library
 Version: {- $OpenSSL::safe::installdata::VERSION -}
-Libs: -L${libdir} -lcrypto
-Libs.private: {- join(' ', @OpenSSL::safe::installdata::LDLIBS) -}
+Libs: -L${libdir} -lcrypto {- join(' ', @OpenSSL::safe::installdata::LDLIBS) -}
 Cflags:{- $OUT = ' -I${includedir}';
           if (scalar @OpenSSL::safe::installdata::INCLUDEDIR_REL_PREFIX > 1) {
               $OUT = '';
--- exporters/pkg-config/libssl.pc.in
+++ exporters/pkg-config/libssl.pc.in
@@ -15,8 +15,7 @@
 Name: OpenSSL-libssl
 Description: Secure Sockets Layer and cryptography libraries
 Version: {- $OpenSSL::safe::installdata::VERSION -}
-Requires.private: libcrypto
-Libs: -L${libdir} -lssl
+Libs: -L${libdir} -lssl -lcrypto
 Cflags:{- $OUT = ' -I${includedir}';
           if (scalar @OpenSSL::safe::installdata::INCLUDEDIR_REL_PREFIX > 1) {
               $OUT = '';
